### PR TITLE
added a `progress` argument to filter the results of `progress()`

### PR DESCRIPTION
**Pull request by tjmahr**
# Summary

I added a `progress` argument to filter the results of `progress()`.

# Related GitHub issues and pull requests

- Ref: #743 

# Checklist

- [x] I have read `drake`'s [code of conduct](https://github.com/ropensci/drake/blob/master/CONDUCT.md), and I agree to follow its rules.
- [x] I have listed any substantial changes in the [development news](https://github.com/ropensci/drake/blob/master/NEWS.md).
- [ ] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/ropensci/drake/tree/master/tests/testthat) to confirm that any new features or functionality work correctly.
- [ ] I have tested this pull request locally with `devtools::check()`
- [x] This pull request is ready for review.
- [] I think this pull request is ready to merge.

An unrelated x86 test failed on my Windows machine.

### CODE REVIEW

File: R/api-progress.R [hunk 1]
```diff
@@ -64,14 +69,23 @@ progress <- function(
...
     targets,
     get_progress_single,
     cache = cache,
     FUN.VALUE = character(1)
   )
-  out <- weak_tibble(target = targets, progress = progress)
+  out <- weak_tibble(target = targets, progress = progress_results)
   rownames(out) <- NULL
+
+  if (!is.null(progress)) {
```
Comment (wlandau) [inline R/api-progress.R]: Thanks @tjmahr! Could we return early if `progress` is `NULL`?

```r
if (is.null(progress)) {
  return(out)
}
progress <- match.arg(
  progress,
  choices = c("done", "running", "failed", "none"),
  several.ok = TRUE
)
out <- out[out$progress %in% progress, ]
```

File: tests/testthat/test-cache.R [hunk 2]
```diff
@@ -484,3 +484,22 @@ test_with_dir("loadd() does not load imports", {
...
 })
+
+test_with_dir("can filter progress", {
+  skip_on_cran() # CRAN gets whitelist tests only (check time limits).
+  plan <- drake_plan(a = TRUE, b = TRUE, c = stop())
+
+  expect_error(make(plan))
+
+  expect_equal(nrow(progress()), 3)
+  expect_equal(nrow(progress(progress = c("done", "failed"))), 3)
```
Comment (wlandau) [inline tests/testthat/test-cache.R]: Could we check the values of `progress()` too? e.g.

```r
out <- progress(x, y, z)
exp <- weak_tibble(target = c("x", "y", "z", progress = c("done", "failed", "none"))
expect_equivalent(out, exp)
```

Comment (tjmahr) [reply tests/testthat/test-cache.R]: done

File: R/api-progress.R (hunk 1 above)
Comment (tjmahr) [reply R/api-progress.R]: done

**Comment by wlandau** [comment]
Thanks @tjmahr!

**Pull request merged**